src/basic/architecture.h

```diff
@@ -49,6 +49,54 @@ typedef enum {
 
 Architecture uname_architecture(void);
 
+/* CPU plus userspace ABI definitions. In most cases you'll want to use this, not Architecture, which is
+ * insufficient for every case apart from the very basics of comparing a running CPU with another. Never
+ * use Architecture when there is any connection with images, binaries, libraries, paths, etc. */
+typedef enum {
+        ABI_ALPHA,
+        ABI_ARC,
+        ABI_ARC_BE,
+        ABI_ARM,
+        ABI_ARMEL,
+        ABI_ARMHF,
+        ABI_ARM64,
+        ABI_ARM64_BE,
+        ABI_ARM_BE,
+        ABI_ARMEL_BE,
```

Comment:
i thought ARMEL stands for "ARM endian little", but what is a BE version of that even supposed to be